tastyworks/tastyworks_api/tw_api.py

```diff
@@ -0,0 +1,647 @@
+    """
+
+    if is_dry_run:
+        url = make_api_url('ORDER_DRY_RUN', account=account_number, order_id=order_id)
+    else:
+        url = make_api_url('ORDER_EXECUTE', account=account_number, order_id=order_id)
+
+    if order_id:
+        resp = await api_request('PATCH', url, token, json_data=order_json)
+    else:
+        resp = await api_request('POST', url, token, json_data=order_json)
+    return resp
+
+
+async def cancel_order(token: str, account_number: str, order_id: str = None) -> Dict:
+    """
+    Cancels an open order defined by "order_id"
+    Args:
+        token (str): A valid session token
+        account_number (str): Account number to use
+        order_id (str): An order 'id' read from an open orders list received via get_orders() or get_orders_live().
+                        It is an optional input parameter to place a new order.
+                        This is required to modify an order for dry run or order routing.
+
+    Returns:
+        The response from the server after canceling the order
+    """
+    # TODO: What is the request and response for the "CANCEL ALL ORDERS" button in the TW platform?
+    #       Could not test on live account
+
+    url = make_api_url('ORDER_EXECUTE', account=account_number, order_id=order_id)
+    resp = await api_request('DELETE', url, token)
+    return resp
+
+
+async def fifty_percent_pop(token: str, pop_json: dict) -> Dict:
+    """
+    Returns the response of the API server including statistical analysis data of a simulated order.
+    JSON data include parameters that needs to be retrieved via market metrics and other API calls.
+    """
+    url = make_api_url('FIFTY_POP')
+    resp = await api_request('POST', url, token, json_data=pop_json)
+    return resp
+
+
+"""
+################################
+######## TRADING DATA ##########
+################################
+"""
+
+
+async def symbol_search(token: str, symbol: str) -> Dict:
+    """
+    Performs a symbol search using Tastyworks API.
+
+    This returns a list of symbols that are similar to the symbol passed in the parameters. This does not provide
+    any details except the related symbols and their descriptions.
+
+    Args:
+        token (str): A valid session token
+        symbol (string): A base symbol (or first characters of a symbol) to search for symbols starting with the string
+
+    Returns:
+        The response from the API server including a list of dictionaries containing the symbols that start with the
+        passed symbol string and a descriptions for each of the found symbols, i.e.
+            [{'symbol': 'K', 'description': 'Kellogg Company Common Stock'},
+            {'symbol': 'KB', 'description': 'KB Financial Group Inc'}]
+    """
+    url = make_api_url('SYMBOL_SEARCH', symbol=symbol)
+    resp = await api_request('GET', url, token)
+    return resp
+
+
+async def get_market_metrics(token: str, symbol_list: list) -> Dict:
```

Comment:
Market Metrics!!  I don't know why I like this endpoint so much, probably hits home to me for some reason.  I don't think this is covered yet!